modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java

```diff
@@ -104,95 +105,220 @@ private void flattenPaths(OpenAPI openAPI) {
     }
 
     /**
-     * Flatten inline models in RequestBody
+     * Return false if model can be represented by primitives e.g. string, object 
+     * without properties, array or map of other model (model contanier), etc.
      *
-     * @param openAPI target spec
-     * @param pathname target pathname
-     * @param operation target operation
+     * Return true if a model should be generated e.g. object with properties,
+     * enum, oneOf, allOf, anyOf, etc.
+     *
+     * @param schema target schema
      */
-    private void flattenRequestBody(OpenAPI openAPI, String pathname, Operation operation) {
-        RequestBody requestBody = operation.getRequestBody();
-        if (requestBody == null) {
-            return;
+    private boolean isModelNeeded(Schema schema) {
```

Comment:
Would is be simpler to make a ref to the current schema, unalias the ref schema, and return True if the unaliased schema still has a ref in it (meaning that a model would be made?
For example:
```
Schema refSchema = new Schema();
refSchema.set$ref("#/components/schemas/"+name);
Schema unaliasedSchema = config.unaliasSchema(refSchema, config.importMapping());
if (unaliasedSchema.get$ref() != null) {
    return true;
}
return false;
```

Reply:
There is [an instance](https://github.com/OpenAPITools/openapi-generator/blob/5bd338f8ee3f22983d66b9f5fbca6699987043e5/modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java#L152) where we don't actually want to unalias the schema but just need to check whether a complex model exists so I'd like to re-use that logic and also I'm not a fan of functions that do unintuitive/unexpected things like in this case checking whether a model is needed AND unaliasing if so. It would save 4 lines of code but doesn't make it "simpler" in my opinion.

Reply:
Thank you for pointing out that use case.
In that case you could call unaliasedSchema directly because the source schema already has a ref. Then if the unaliased schema had a ref in it, we know that the model would be produced.

> we don't actually want to unalias the schema but just need to check whether a complex model exists

I agree. But we can use the results of the unaliased schema to tell if a model will be made.
A comment could explain why we are using unaliasSchema to tell if a model is going to be made, making this a more intuitive usage.
My suggestion was not to just replace one block with my pseudocode above, it was to replace the entire body of isModelNeeded with that block. This would reduce the size of the method by 16 lines. It's you call though.